src/snes/supermetroid/modules/RidleyOverkillTracker.js

```diff
@@ -0,0 +1,139 @@
+            Addresses.ridleyIframeTimer,
+            Addresses.ridleyShotDamage,
+        ];
+    }
+
+    reset() {
+        this.lastCheck = 0;
+        this.inRidleyFight = false;
+        this.startCounting = false;
+        this.additionalShotCounter = 0;
+        this.additionalDamageDealt = 0;
+    }
+
+    getMessage() {
+        return 'That was an overkill. ' +
+            'You had ' + this.additionalShotCounter + ' ' +
+            'additional <{plural("shot", ' + this.additionalShotCounter + ')}>'
+    }
+
+    memoryReadAvailable({ memory, sendEvent }) {
+        // Handle a run being reset
+        if (
+            memory.roomID.prevFrameValue !== undefined &&
+            memory.roomID.prevFrameValue !== Rooms.EMPTY &&
+            memory.roomID.value === Rooms.EMPTY
+        ) {
+            if (this.inRidleyFight) {
+                // were in ridley fight but reset due to ridley
+            }
+            this.reset();
+            return;
+        }
+
+        // if we are not yet in the fight, but we are in ridley room and he appears
+        if (
+            !this.inRidleyFight &&
+            memory.roomID.value === Rooms.LowerNorfair.RIDLEY_ROOM &&
+            this.checkChange(memory.enemyHP)
+        ) {
+            if (memory.enemyHP.value !== 0) {
+                this.inRidleyFight = true;
+                return;
+            }
+        }
+
+        // we are in ridley fight
+        else if (this.inRidleyFight) {
+
+            // we left ridley room in whatever direction
+            if (this.checkTransition(memory.roomID, Rooms.LowerNorfair.RIDLEY_ROOM)) {
+                if (this.additionalShotCounter > 0) {
+                    if (this.settings.displayDoorTimes.value) {
+                        sendEvent('ridleyOverkill', {shots:this.additionalShotCounter, damage:this.additionalDamageDealt});
+                    }
+                    else {
+                        sendEvent('msg', this.getMessage(), 3);
+                    }
+                }
+                this.reset();
+            }
+
+            // keep watching for the last hit
+            else if (memory.enemyHP.value <= 1000 && this.checkTransition(memory.enemyHP, undefined, 0)) {
+                this.startCounting = true;
+
+                // make sure to do next check only after 10 frames
+                this.lastCheck = Date.now();
+            }
+
+            // he is dead
+            else if (memory.enemyHP.value === 0 && this.startCounting) {
+
+                const time = Date.now();
+
+                // only check every 10 frames
+                if (time - this.lastCheck > (1000 / 60 * 10))  {
+                    if (memory.ridleyIframeTimer && memory.ridleyIframeTimer.value === 10) {
+                        this.additionalShotCounter++;
+
+                        if (memory.ridleyShotDamage && memory.ridleyShotDamage.value > 0) {
+                            // this is not reliable, somehow this is below 900 sometimes
```

Comment:
I think there may be a few issues here: for one, I believe the timer starts counting at hex value $10 which is decimal 16 rather than decimal value 10. You may want to confirm that though.

The problem is that with our polling rate there is no guarantee that you will see the exact frame where the value is $10, so it's better to do some fuzzy matching here. I don't fully understand why you're only checking every 10 frames rather than watching for the timer to reset back up to $10 or something close to it.